heroku_connect/models.py

```diff
@@ -0,0 +1,342 @@
+
+    def __str__(self):
+        created_at = self.created_at
+        if created_at:
+            created_at = '{:%Y-%m-%d %a %H:%M%z}'.format(created_at)
+        return (
+            '#{id} {action} {table_name}|{record_id} [{created_at}] [{state}]'
+        ).format(id=self.id, action=self.action, table_name=self.table_name,
+                 record_id=self.record_id, created_at=created_at, state=self.state)
+
+    def get_model(self):
+        """Fetch the instance of the connected model referenced by this log record.
+
+        Returns:
+            The connected instance, or ``None`` if it does not exists.
+
+        """
+        model_cls = hc_models.registry.get_class_for_table_name(self.table_name)
+        return model_cls._default_manager.filter(id=self.record_id).first()
+
+    def related(self, *, exclude_self=False):
+        """Get a QuerySet for all trigger log objects for the same connected model.
+
+        Args:
+            exclude_self: Whether to exclude this log object from the result list
+        """
+        manager = type(self)._default_manager
+        queryset = manager.related_to(self)
+        if exclude_self:
+            queryset = queryset.exclude(id=self.id)
+        return queryset
+
+    def capture_insert(self, *, exclude_fields=()):
+        """Apply :meth:`.TriggerLogAbstract.capture_insert_from_model` for this log."""
+        return self.capture_insert_from_model(self.table_name, self.record_id,
+                                              exclude_fields=exclude_fields)
+
+    def capture_update(self, *, update_fields=()):
+        """Apply :meth:`.TriggerLogAbstract.capture_insert_from_model` for this log."""
+        return self.capture_update_from_model(self.table_name, self.record_id,
+                                              update_fields=update_fields)
+
+    @staticmethod
+    def _fieldnames_to_colnames(model_cls, fieldnames):
+        """Get the names of columns referenced by the given model fields."""
+        get_field = model_cls._meta.get_field
+        fields = map(get_field, fieldnames)
+        return {f.column for f in fields}
+
+
+class TriggerLog(TriggerLogAbstract):
+    """Represents entries in the Heroku Connect trigger log.
+
+    .. seealso:: :class:`TriggerLogAbstract`
+    """
+
+    is_archived = False
+
+    class Meta(TriggerLogAbstract.Meta):
+        db_table = '{schema}"."_trigger_log'.format(schema=settings.HEROKU_CONNECT_SCHEMA)
+        verbose_name = 'Current Trigger Log'
+
+
+class TriggerLogArchive(TriggerLogAbstract):
+    """Represents entries in the Heroku Connect trigger log archive.
+
+    .. seealso:: :class:`TriggerLogAbstract`
+    """
+
+    is_archived = True
+
+    class Meta(TriggerLogAbstract.Meta):
+        db_table = '{schema}"."_trigger_log_archive'.format(schema=settings.HEROKU_CONNECT_SCHEMA)
+        verbose_name = 'Archived Trigger Log'
+
+
+class TriggerLogPermanent(TriggerLogAbstract):
```

Comment:
I don't know about this one. I don't think Postgres is a good place for logging data to begin with and even less as an archive. I like the idea of retaining them, but only via `syslog`. Users can decide than if you want to persist these logs. EG. we use S3 to persist logs over 30 days old.